datafusion/core/tests/physical_optimizer/filter_pushdown/util.rs

```diff
@@ -61,6 +62,12 @@ impl FileOpener for TestOpener {
         _file_meta: FileMeta,
         _file: PartitionedFile,
     ) -> Result<FileOpenFuture> {
+        if let Some(predicate) = &self.predicate {
```

Comment:
Will take this out, just leaving here for now for visualization.

I wonder what are good ways to regression test this 🤔

Reply:
The best I can think of would be to make a custom ExecutionPlan that delays partition 0 for 1s or something and check that we still read the expected number of rows on the probe side

Reply:
Hey @adriangb, I modified the existing partitioned test to inspect metrics from the probe side scan and verify rows are correctly being filtered out there. I think we can get away without the delay since I applied the patch to main and it was flaky as expected. Does that sound good to you or did you have something else in mind?